extensions/extension.go

```diff
@@ -174,10 +202,36 @@ func evalFunc(env any, name string, args []object.Object) object.Object {
 	return res
 }
 
+// Normalizes to alphanum.gr.
+func sanitizeFileName(file string) (string, error) {
+	if emptyOnly && file != "" {
+		return "", fmt.Errorf("empty only mode, filename must be empty or no arguments, got: %q", file)
+	}
+	if unrestrictedIOs {
+		log.Infof("Unrestricted IOs, not sanitizing filename: %s", file)
+		return file, nil
+	}
+	// only alhpanumeric and _ allowed. no dots, slashes, etc.
+	f := strings.TrimSuffix(file, ".gr")
+	for _, r := range []byte(f) {
+		if !lexer.IsAlphaNum(r) {
+			return "", fmt.Errorf("invalid character in filename %q: %c", file, r)
+		}
+	}
+	return f + ".gr", nil
```

Comment:
Why not simply returning


```suggestion
	return file, nil
```

Reply:
because .gr is optional in the input (ie loads/save("foo") is valid (and will end up using foo.gr))

Reply:
Indeed, then. You trim if present, then add it